src/runtime/hashmap.h

```diff
@@ -6,6 +6,14 @@
 #include "printer.h"
 #include "scoped_mutex_lock.h"
 
+#ifndef hashmap_malloc
+#define hashmap_malloc(size)    halide_malloc(NULL, size)
+#endif//hashmap_malloc
+
+#ifndef hashmap_free
+#define hashmap_free(memory)    halide_free(NULL, memory)
+#endif//hashmap_free
+
```

Comment:
I'm not sure I understand why this change; is everything the same as before, except now there's an extra preprocessor definition?

Reply:
Yes. This way, whichever runtime module that needs to include `hashmap.h` has a chance to override the default malloc/free implementation.
That's what `d3d12compute.cpp` is doing:
https://github.com/halide/Halide/blob/dbf75159454a6380a41eef7e2ee90922919ee423/src/runtime/d3d12compute.cpp#L200-L202

Reply:
I added some comments above it explaining the rationale of it.